dali/test/python/operator_2/test_random_bbox_crop.py

```diff
@@ -152,15 +152,55 @@ def crop_contains(crop_anchor, crop_shape, point):
 
 def filter_by_centroid(crop_anchor, crop_shape, bboxes):
     ndim = len(crop_shape)
-    nboxes = bboxes.shape[0]
     indexes = []
-    for i in range(nboxes):
-        bbox = bboxes[i]
+    for i, bbox in enumerate(bboxes):
         centroid = [0.5 * (bbox[d] + bbox[ndim + d]) for d in range(ndim)]
         if crop_contains(crop_anchor, crop_shape, centroid):
             indexes.append(i)
-        filtered_boxes = np.array(bboxes[indexes, :])
-    return filtered_boxes
+    return np.array(bboxes[indexes, :])
+
+
+def intersection(bbox_a: np.ndarray, bbox_b: np.ndarray):
+    """bbox_a and bbox_b are xy(z)XY(Z) format"""
+    ndim, rem = divmod(len(bbox_a), 2)
+    assert rem == 0, "Should be even dims"
+    xy = np.maximum(bbox_a[:ndim], bbox_b[:ndim])
+    XY = np.minimum(bbox_a[ndim:], bbox_b[ndim:])
+    sides = XY - xy
+    return 0 if any(sides < 0) else np.prod(sides)
+
+
+def test_intersection():
+    a = np.array([0, 0, 2, 2])
+    b = np.array([2, 2, 4, 4])
+    h = np.array([3, 3, 4, 4])
+    assert intersection(a, h) == 0
+    assert intersection(a, b) == 0
+    assert intersection(b, a) == 0
+    assert intersection(b, b) == 4
+    c = np.array([1, 1, 4, 4])
+    assert intersection(a, c) == 1
+    assert intersection(b, c) == 4
+    d = np.array([0, 0, 2, 3])
+    e = np.array([1, 1, 3, 2])
+    assert intersection(d, e) == 1
+    assert intersection(e, d) == 1
+    f = np.array([0, 0, 0, 3, 3, 3])
+    g = np.array([1, 1, 1, 4, 4, 4])
+    assert intersection(f, g) == 8
+
+
+def filter_by_area(crop_anchor, crop_shape, bboxes, thresh):
+    ndim = len(crop_shape)
+    crop_box = np.concatenate([crop_anchor, crop_shape])  # xywh
+    crop_box[ndim:] += crop_box[:ndim]  # xyXY
+    indexes = []
+    for i, bbox in enumerate(bboxes):
+        intersec = intersection(bbox, crop_box)
+        box_area = np.prod(bbox[ndim:] - bbox[:ndim])
+        if intersec / box_area > thresh:
```

Comment:
```suggestion
        if intersec / box_area >= thresh:
```
I guess the tests need to be updated as well.